samza-api/src/main/java/org/apache/samza/table/descriptors/RemoteTableDescriptor.java

```diff
@@ -345,7 +346,6 @@ public String getProviderFactoryClassName() {
 
   @Override
   protected void validate() {
-    Preconditions.checkNotNull(readFn, "TableReadFunction is required.");
```

Comment:
Thanks for the change!

should we validate at least a readfn or writefn exist? 
Also, The constructor of RemoteTable and AsyncRemoteTable also has validations that readfn is required. Purely remove the validation here may not be enough. 

After this change, it's possible unit/regression test may fail, need to fix as well if any.

Reply:
Good catch, added!